### ISSUE TITLE
Event logging says Invalid Signature passed in Authorization

### ISSUE THREAD

**Issue by dineshiwiz** [Azure/diagnostics-eventflow#170]
{"Error":"InvalidAuthorization","Message":"An invalid signature was specified in the Authorization header"}

**Comment by karolz-ms** [comment]
Roger. Investigating...

**Comment by dineshiwiz** [comment]
thanks Karolz, I am really stuck at this point since once this error occurs the cluster node starts displaying this error on service fabric explorer.

**Comment by karolz-ms** [comment]
@dineshiwiz I did some poking around and I did not find anything obviously wrong.

Some users reported similar issues with the OMS data collector API on the forums: https://social.msdn.microsoft.com/Forums/azure/en-US/3bdbb610-b45f-46cb-bb54-77d820e6c04d/oms-datacollector-api-using-httpclient-in-c-authorization-failed?forum=opinsights I double-checked EventFlow code and we already follow the recommended practice. Also, I believe the error would occur predictably if it was a code mistake in the way we construct the signature.

Is it possible that sometimes your diagnostic data exceeds the OMS data collector API data limits? https://docs.microsoft.com/en-us/azure/log-analytics/log-analytics-data-collector-api#data-limits If this is a possibility, you might be able to reduce the size of the data by reducing the maximum event batch size. See https://github.com/Azure/diagnostics-eventflow#pipeline-settings for details and let me know if this helps!

**Comment by dineshiwiz** [comment]
I will try your suggestions and update you. Thanks @karolz-ms for the update.

**Comment by dineshiwiz** [comment]
Hi @karolz-ms , I am seeing this logs under  Operation category in log analysis OMS Portal

"
The following fields' values Payload_Message,Payload_Response of type Event have been trimmed to the max allowed size, 32766 bytes. Please adjust your input accordingly.
--
"

do you think this may have caused this error on my cluster explorer?

**Comment by karolz-ms** [comment]
@dineshwiz I think so, yes. I do not know the details of the OMS data collector API implementation, but if this trimming happens before the signature is validated, that would explain the invalid signature error reported by the API

Do you think there is a way for you to reduce the size of the Message and Response properties of your traces?

**Comment by dineshiwiz** [comment]
Hi @karolz-ms it did fix the issue, once i reduced the size, though i am seeing another error on my service fabric cluster explorer:
"An error occurred while sending data to OMS: System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Diagnostics.EventFlow.Outputs.HttpExponentialRetryMessageHandler.d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Net.Http.HttpClient.d__58.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Diagnostics.EventFlow.Outputs.OmsOutput.d__9.MoveNext()"


Could you please tell me the reason for this? and why does it affect my cluster?

**Comment by karolz-ms** [comment]
Glad that original issue was resolved!

The exception you are seeing is probably harmless--it can happen when the service is asked to shut down (either because of human administrative action, or because of Service Fabric re-balancing the work across the cluster). It should not really be treated as a problem, so it is a bug that it is reported as such. I have opened [another issue](https://github.com/Azure/diagnostics-eventflow/issues/172) to have this fixed. ETA sometime next week.

**Comment by stevieboy84** [comment]
We get this issue on our clusters. It happens on random nodes.  A side effect of this happening, given its transient nature, is that sometimes on deployments, we get failures due to unhealthy application evaluations.  The last couple of production deployments we did I ended up having to delete the service before deploying, so that we didn't do a rolling upgrade, which is far from ideal.  I'll try what @karolz-ms  suggested with reducing the size and see if that resolves it.
